src/dg/dg.cpp

```diff
@@ -1474,16 +1502,22 @@ void DGBase<dim,real,MeshType>::assemble_residual (const bool compute_dRdW, cons
                 flux_basis_int,
                 flux_basis_ext,
                 flux_basis_stiffness,
+                soln_basis_projection_oper_int,
+                soln_basis_projection_oper_ext,
                 mapping_basis,
                 false,
                 right_hand_side,
                 auxiliary_right_hand_side);
         } // end of cell loop
+
+        timer.stop();
+        assemble_residual_time += timer.cpu_time();
```

Comment:
Should we consider adding a parameter to check for the time only if we are doing timing runs. If its a production run, wouldn't we be introducing a mpi_waitall to capture the timing?

Reply:
Yes we can add a parameter. 
From what I gathered timer.cpu_time() stores the local cpu time for the processor because it was not initialized with the mpi_communicator (line 1471). It does not wait for the other processors to reach the point to store the time. The mpi sum happens in the test after the entire run and that's when the mpi_waitall would be called. If it was initialized like: dealii::Timer timer(this->mpi_communicator,false) then yes it would have an mpi_waitall within it when cpu_time() get called.